### ISSUE TITLE
Allow serialization of Galois field

### ISSUE THREAD

**Issue by raeudigerRaeffi** [mhostetter/galois#388]
I recently tried playing around with multithreading and noticed, that I can not pass the Galois field object to a thread, since the Galois class can not be pickled. An example error would be
PicklingError: Can't pickle <class 'galois.GF(47)'>: attribute lookup GF(47) on galois failed

**Comment by raeudigerRaeffi** [comment]
would probably be a feature request

**Comment by mhostetter** [comment]
I don't believe I've made a class pickleable before, but it seems possible. I think maybe adding `__get_state__()` and `__set_state__()` to `FieldArrayMeta` will do the trick. I'll look into it and play around.

https://docs.python.org/3/library/pickle.html#pickling-class-instances

**Comment by mhostetter** [comment]
@raeudigerRaeffi I believe I resolved this in #393. Can you test it on your side?

You can install that branch like this.

```
$ python3 -m pip uninstall galois
$ python3 -m pip install git+https://github.com/mhostetter/galois@support-pickle
```

**Comment by mhostetter** [comment]
@raeudigerRaeffi this should be resolved. Feel free to re-open or open a new issue if you continue to have problems.
